backend/stakepoold/rpcclient.go

```diff
@@ -110,70 +113,201 @@ func connectWalletRPC(cfg *config) (*dcrrpcclient.Client, semver, error) {
+	// Create raw transaction.
+	var buf []byte
+	buf, err := hex.DecodeString(txhex)
+	if err != nil {
+		return false, fmt.Errorf("DecodeString failed: %v", err)
+	}
+	res, err := ctx.nodeConnection.DecodeRawTransaction(buf)
+	if err != nil {
+		return false, fmt.Errorf("DecodeRawTransaction failed: %v", err)
+	}
+
+	bhash, err := chainhash.NewHashFromStr(blockhash)
+	if err != nil {
+		return false, fmt.Errorf("NewHashFromStr failed for %v: %v", blockhash, err)
+	}
+	gbh, err := ctx.nodeConnection.GetBlockHeader(bhash)
+	if err != nil {
+		return false, fmt.Errorf("GetBlockHeader failed for %v: %v", bhash, err)
+	}
+
+	// Check the first commitment output (Vout[1])
+	// and ensure that the address found there exists
+	// in the list of approved addresses. Also ensure
+	// that the fee exists and is of the amount
+	// requested by the pool.
+
+	commitAmt := dcrutil.Amount(0)
+	feeAddr := ""
+	feeAddrValid := false
+	ticketValue, err := dcrutil.NewAmount(res.Vout[0].Value)
+	if err != nil {
+		return false, fmt.Errorf("NewAmount failed: %v", err)
+	}
+
+	for i := range res.Vout {
+		switch res.Vout[i].ScriptPubKey.Type {
+		case "sstxcommitment":
+			for j := range res.Vout[i].ScriptPubKey.Addresses {
+				_, exists := ctx.feeAddrs[res.Vout[i].ScriptPubKey.Addresses[j]]
+				if !exists {
+					continue
+				}
+				feeAddr = res.Vout[i].ScriptPubKey.Addresses[j]
+				feeAddrValid = true
+				feeAmount, err := dcrutil.NewAmount(*res.Vout[i].ScriptPubKey.CommitAmt)
+				if err != nil {
+					return false, fmt.Errorf("NewAmount failed: %v", err)
+				}
+				commitAmt = feeAmount
+			}
+		}
+	}
+
+	in := dcrutil.Amount(0)
+	for i := range res.Vout {
+		switch res.Vout[i].ScriptPubKey.Type {
+		case "sstxcommitment":
+			amount, err := dcrutil.NewAmount(*res.Vout[i].ScriptPubKey.CommitAmt)
+			if err != nil {
+				return false, fmt.Errorf("NewAmount failed: %v", err)
+			}
+			in += amount
+		}
+	}
+
+	out := dcrutil.Amount(0)
+	for i := range res.Vout {
+		amount, err := dcrutil.NewAmount(res.Vout[i].Value)
+		if err != nil {
+			return false, fmt.Errorf("NewAmount failed: %v", err)
+		}
+		out += amount
+	}
+	fees := in - out
+
+	if !feeAddrValid {
+		log.Warnf("Unknown pool commitment address %s for ticket %v",
+			feeAddr, res.Txid)
+		return false, nil
+	}
+
+	// Calculate the fee required based on the current
+	// height and the required amount from the pool.
+	feeNeeded := txrules.StakePoolTicketFee(ticketValue, fees,
+		int32(gbh.Height), ctx.feePercent, ctx.params)
+	if commitAmt < feeNeeded {
+		log.Warnf("User %s submitted ticket %v which has less fees than "+
+			"are required to use this stake pool and is being skipped "+
+			"(required: %v, found %v)", feeAddr, res.Txid, feeNeeded, fees)
+
+		// Reject the entire transaction if it didn't
+		// pay the pool server fees.
+		return false, nil
+	}
+
+	log.Debugf("Accepted valid stake pool ticket %v committing %v in fees",
+		res.Txid, fees)
```

Comment:
This is printing the transaction/mining fee.  Is the intent to print the commitAmt?